src/middlewared/middlewared/plugins/device_/vrrp_events.py

```diff
@@ -59,26 +60,26 @@ def run(self):
                     sleep(self._retry_timeout)
 
 
+def start_or_stop_vrrp_thread(middleware, licensed, shutting_down):
+    try:
+        vthread = [i for i in threading.enumerate() if i.name == VRRP_THREAD_NAME][0]
```

Comment:
It's a bit unusual to identify threads by their names and to use the thread name to ensure it is not already running. The classic approach is to just have a global variable with this thread (and start/stop thread and assign this variable using a shared lock to avoid race conditions).

Reply:
I actually see this was approach that was used before, what's wrong with that?

Reply:
If you move managing this thread to a plugin, you won't even need to access the thread variable from other modules

Reply:
yeah agree, that was the biggest problem with the way I was doing it previously